### PR TITLE
feat: Add results visualization tool for GCP scanner

### PR DESCRIPTION
**Pull request by yahia3200**
## Description

Merge the visualization tool into the main branch

## Todo
- [x] Fix merge conflicts
- [x] Add visualization tool build files to git ignore
- [x] Include visualization tool files in the `MANIFEST.in`
- [x] Update `INSTALL.md` to include the build step for the visualization tool when building from source
- [x] Add a build step for the visualization tool in the `python-publish` workflow

### CODE REVIEW

**Comment by mshudrak** [comment]
Sorry for the delay. I will make sure to take a look today and tomorrow.

File: visualization_tool/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+You can also remove any file by clicking on the `minus` icon next to the file name.
+
+## Viewing the results
+
+There are two main pages in the tool:
+
+### Resources Page
+
+This main page and it shows the list of resources that were found in the result file. You can click on the `Details` button to see more details about the resource.
```
Comment (mshudrak) [inline visualization_tool/USAGE.md:29]: `This main page` replace with `This is the main page`

File: visualization_tool/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+![GCP Scanner Visualizer](../misc/visualization_tool_images/details.webp)
+
+#### Sorting and filtering resources
+
+You can sort the shown resources by their name or creation time. by default, the resources are sorted by their creation date in descending order and you can change the sorting type from the `Sort` Section on the left menu.
```
Comment (mshudrak) [inline visualization_tool/USAGE.md:35]: `by default` replace with `By default`

File: visualization_tool/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+#### Searching for resources
+
+You can search for resources by their name in the search bar. The tool will show the resources that contain the search term in their name.
+
+![GCP Scanner Visualizer](../misc/visualization_tool_images/search_resources.webp)
+
+### IAM Policy Page
+
+This page shows the list of IAM policies that were found in the result file. The policies are visualized in a table view, which each row represents a single role and the roles names prefixed with the project name.
```
Comment (mshudrak) [inline visualization_tool/USAGE.md:54]: replace `, which` with `with` (don't forget to remove comma)

File: visualization_tool/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "gcp-scanner",
+  "private": true,
+  "version": "0.0.0",
```
Comment (mshudrak) [inline visualization_tool/package.json:4]: May be say at least 1.0.0? :)

**Review by mshudrak — COMMENTED** [review]
I left a couple of minor comments about grammar but other than that it should be almost ready to get merged.

File: visualization_tool/USAGE.md (hunk 1 above)
Comment (yahia3200) [reply visualization_tool/USAGE.md]: Done

File: visualization_tool/USAGE.md (hunk 2 above)
Comment (yahia3200) [reply visualization_tool/USAGE.md]: Done

File: visualization_tool/USAGE.md (hunk 3 above)
Comment (yahia3200) [reply visualization_tool/USAGE.md]: Done

File: visualization_tool/package.json (hunk 4 above)
Comment (yahia3200) [reply visualization_tool/package.json]: Done

**Comment by mshudrak** [comment]
LGTM

**Review by mshudrak — APPROVED** [review]
LGTM

**Pull request merged**